src/overlap_pmmg.c

```diff
@@ -0,0 +1,736 @@
+  }
+
+  /* Loop over the number of external node communicator */
+  for (icomm=0; icomm<next_comm; icomm++) {
+
+    /* Local initialization */
+    LOCAL_MPI_TAG     = 1;
+    npPBDY_in2out     = 0;
+    npInterior_in2out = 0;
+    npTot_in2out      = 0;
+    ntTot_in2out      = 0;
+    ndataPBDY_in2out  = 0;
+    np_in             = mesh->np;
+
+    /* Get external node communicator information */
+    ext_comm  = &parmesh->ext_node_comm[icomm]; // External node communicator
+    color_in  = ext_comm->color_in;             // Color of this partition Pcolor_in
+    color_out = ext_comm->color_out;            // Color of the remote partition Pcolor_out
+    nitem_ext = ext_comm->nitem;                // Nbr of nodes in common between Pcolor_in and Pcolor_out
+
+    /* Overlap variables */
+    overlap = &parmesh->overlap[icomm];
+    overlap->color_in  = color_in;
+    overlap->color_out = color_out;
+
+    /* Local allocation memory */
+    PMMG_CALLOC(parmesh,n_ToSend,6,int,"n_ToSend",ier = 0);
+    PMMG_CALLOC(parmesh,pointCoordInterior_ToSend, 3*mesh->np, double,  "pointCoordInterior_ToSend",ier = 0);
+    PMMG_CALLOC(parmesh,pointCoordPBDY_ToSend,     3*mesh->np, double,  "pointCoordPBDY_ToSend",    ier = 0);
+    PMMG_CALLOC(parmesh,pointTagInterior_ToSend,     mesh->np, uint16_t,"pointTagInterior_ToSend",  ier = 0);
+    PMMG_CALLOC(parmesh,pointTagPBDY_ToSend,         mesh->np, uint16_t,"pointTagPBDY_ToSend",      ier = 0);
+    PMMG_CALLOC(parmesh,pointRefInterior_ToSend,     mesh->np, int,     "pointRefInterior_ToSend",  ier = 0);
+    PMMG_CALLOC(parmesh,pointRefPBDY_ToSend,         mesh->np, int,     "pointRefPBDY_ToSend",      ier = 0);
+    PMMG_CALLOC(parmesh,pointIdxPBDY_ToSend,         mesh->np, int,     "pointIdxPBDY_ToSend",      ier = 0);
+    PMMG_CALLOC(parmesh,pointIdxInterface_ToSend,    nitem_ext,int,     "pointIdxInterface_ToSend", ier = 0);
+    PMMG_CALLOC(parmesh,tetraVertices_ToSend,      4*mesh->ne, int,     "tetraVertices_ToSend",     ier = 0);
+    PMMG_CALLOC(parmesh,tetraVerticesSeen_ToSend,  4*mesh->ne, int,     "tetraVerticesSeen_ToSend", ier = 0);
+    PMMG_CALLOC(parmesh,tetraRef_ToSend,             mesh->ne, int,     "tetraRef_ToSend",          ier = 0);
+    PMMG_CALLOC(parmesh,lsInterior_ToSend,           mesh->np, double,  "lsInterior_ToSend",        ier = 0);
+    PMMG_CALLOC(parmesh,lsPBDY_ToSend,               mesh->np, double,  "lsPBDY_ToSend",            ier = 0);
+
+    /** STEP 1 - Identify nodes at the interface between Pcolor_in and Pcolor_out
+              a) Assign flag -1 to these nodes
+              b) Store the local point indices in pointIdxInterface_ToSend
+                 to be able to fill hash_out2in **/
+    /* Loop over the nodes in the external node communicator */
+    for (i=0; i < nitem_ext; i++) {
+      /* Get the index of the node in internal communicator */
+      idx = ext_comm->int_comm_index[i];
+      ip  = int_comm->intvalues[idx];
```

Comment:
Ok. The comment is now:
`
Note: The following works because we have only one group in this function
          (see the assert at the beginning of the function) and we have been able
          to store the local index of node
`